service/notesservice.php

```diff
@@ -196,6 +181,31 @@ public function delete ($id, $userId) {
         $file->delete();
     }
 
+    private function getSafeTitleFromContent($content) {
+        // prepare content: remove markdown characters and empty spaces
+        $content = preg_replace("/^\s*[*+-]\s+/m", "", $content); // list item
+        $content = preg_replace("/^#+\s+(.*?)\s*#*$/m", "$1", $content); // headline
+        $content = preg_replace("/^(?:=*|-*)$/m", "", $content); // separate line for headline
```

Comment:
Why do you explicitly use a non-capturing group? Does this have any advantages as opposed to `/^(=*|-*)$/m`.

Also shouldn't we write this as: `/^(=+|-+)$/m`? Otherwise empty lines match. Or was this intended?